### ISSUE TITLE
requestPermission not working when "ACCESS_BACKGROUND_LOCATION" is given

### ISSUE THREAD

**Issue by thealteria** [Baseflow/flutter-geolocator#771]
I am developing an app which uses background location. When I'm trying to request the location permission via `requestPermission`, it's not showing me the request permission prompt. It's working fine if I don't put the `ACCESS_BACKGROUND_LOCATION` permission in the manifest file.

**Comment by mvanbeusekom** [comment]
@thealteria just to give you a fair warning, at the moment the geolocator plugin doesn't support accessing the location services in the background. Meaning when your app moves to the background (especially on Android) the location services will stop and not continue to supply updates.

It does however support requesting permission to access the permission in the background, on Android API 29 and higher you will need to add the `ACCESS_BACKGROUND_LOCATION` together with the already existing `ACCESS_FINE_LOCATION` permission. So you need to add both permissions to your AndroidManifest.xml file.

We are planning to add support for background location but this will not happen on the short term. Will close this issue now, if you have a question feel free to leave a comment.

**Comment by thealteria** [comment]
I'm not accessing "Background location" from the Geolocator. Please read my issue again. I'm saying that the "request permission prompt" is not coming when  put the `ACCESS_BACKGROUND_LOCATION` in the manifest file.

The `requestPermission` is not working in this case I'm just saying that.

I repeat, I'm not trying to access "location services in the background" from `geolocator`. I'm just requesting the `permission` using `requestPermission` and it's not working.

**Comment by mvanbeusekom** [comment]
I have been doing some tests and it seems this is an issue on Android API 30 and higher (Android API 29 and below works as expected). We will look into this further. 

The files below can be used to reproduce the behaviour:

<details>
<summary>pubspec.yaml</summary>

```yaml
name: geo_issue_768
description: A new Flutter project.
publish_to: 'none'
version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  geolocator: ^7.2.0+1
  app_tracking_transparency: ^2.0.1
  cupertino_icons: ^1.0.2

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true
```
</details>

<details>
<summary>AndroidManifest.xml</summary>

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.geo_issue_768">

   <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
   <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />

   <application
        android:label="geo_issue_768"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            <!-- Specifies an Android theme to apply to this Activity as soon as
                 the Android process has started. This theme is visible to the user
                 while the Flutter UI initializes. After that, this theme continues
                 to determine the Window background behind the Flutter UI. -->
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              android:resource="@style/NormalTheme"
              />
            <!-- Displays an Android View that continues showing the launch screen
                 Drawable until Flutter paints its first frame, then this splash
                 screen fades out. A splash screen is useful to avoid any visual
                 gap between the end of Android's launch screen and the painting of
                 Flutter's first frame. -->
            <meta-data
              android:name="io.flutter.embedding.android.SplashScreenDrawable"
              android:resource="@drawable/launch_background"
              />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        <!-- Don't delete the meta-data below.
             This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>
```
</details>

<details>
<summary>main.dart</summary>

```dart
import 'package:app_tracking_transparency/app_tracking_transparency.dart';
import 'package:flutter/material.dart';
import 'package:geolocator/geolocator.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  void _requestLocationPermissions() {
    Geolocator.requestPermission()
        .then((status) => print('Location permissions: $status'));
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      floatingActionButton: Column(
        mainAxisAlignment: MainAxisAlignment.end,
        children: [
          FloatingActionButton(
            onPressed: _requestLocationPermissions,
            tooltip: 'Locaton Permisssions',
            child: const Icon(Icons.location_pin),
          ),
        ],
      ),
    );
  }
}
```
</details>

**Comment by florissmit1** [comment]
Thanks for submitting this issue, we fixed it in geolocator 7.3.1. `requestPermission` will now behave like expected when `ACCESS_BACKGROUND_LOCATION` is added to the `AndroidManifest.xml` in your app. Hereby I will close this issue.
